src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -139,19 +140,36 @@ export class TerminalInstance implements ITerminalInstance {
 		this._initDimensions();
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
+		if (this._shellLaunchConfig) {
+			this.setTitle(path.basename(this._shellLaunchConfig.executable), true);
+		}
 
 		if (platform.isWindows) {
```

Comment:
This is getting very large, can we instead pass this._xterm to `WindowsShellHelper` and do everything in there? Encapsulating all this logic into `WindowsShellHelper` would be best as it's Windows specific and `TerminalInstance` is already quite large and complex.